testsuite/openshift/objects/rate_limit.py

```diff
@@ -0,0 +1,55 @@
+"""RateLimitPolicy related objects"""
+from time import sleep
+
+from testsuite.openshift.client import OpenShiftClient
+from testsuite.openshift.objects import OpenShiftObject, modify
+from testsuite.openshift.objects.gateway_api import Referencable
+
+
+class RateLimitPolicy(OpenShiftObject):
+    """RateLimitPolicy (or RLP for short) object, used for applying rate limiting rules to an Gateway/HTTPRoute"""
+
+    @classmethod
+    def create_instance(cls, openshift: OpenShiftClient, name, route: Referencable, labels: dict[str, str] = None):
+        """Creates new instance of RateLimitPolicy"""
+        model = {
+            "apiVersion": "kuadrant.io/v1beta1",
+            "kind": "RateLimitPolicy",
+            "metadata": {"name": name, "namespace": openshift.project},
+            "spec": {
+                "targetRef": route.reference,
+                "rateLimits": [
+                    {
+                        "configurations": [
+                            {"actions": [{"generic_key": {"descriptor_key": "limited", "descriptor_value": "1"}}]}
+                        ]
+                    }
+                ],
+            },
+        }
+
+        if labels is not None:
+            model["metadata"]["labels"] = labels  # type: ignore
+
+        return cls(model, context=openshift.context)
+
+    @modify
+    def add_limit(self, max_value, seconds, conditions: list[str] = None):
+        """Add another limit"""
+        conditions = conditions or []
+        limits = self.model.spec.rateLimits[0].setdefault("limits", [])
+        limit = {"maxValue": max_value, "seconds": seconds, "conditions": conditions, "variables": []}
```

Comment:
The doc is currently outdated, however it should be optional since (https://github.com/Kuadrant/kuadrant-operator/pull/144) which we dont have deployed yet. Once we have a build containing the feature, I will update PR.